lib/formats/bs_format_ripejson.c

```diff
@@ -552,10 +599,10 @@ bs_format_ripejson_populate_record(bgpstream_format_t *format,
   int filter;
 
 retry:
-  STATE->json_string_buffer_len = bgpstream_transport_readline(
-    format->transport, STATE->json_string_buffer, BGPSTREAM_PARSEBGP_BUFLEN);
 
-  assert(STATE->json_string_buffer_len < BGPSTREAM_PARSEBGP_BUFLEN);
+  STATE->json_string_buffer_len = readline(format);
+
+  assert(STATE->json_string_buffer_len < STATE->json_string_buffer_alloc_len);
```

Comment:
can the length not be equal to the alloc len?

Reply:
wandio readline forces null-termination, therefore the `buffer_len` is at most `alloc_len-1`.